### DISCUSSION TITLE
GitHub Actions Integration(s)?

### DISCUSSION THREAD

**Discussion by bobcatfish** [tektoncd/community#340] [Ideas]
Hey all!

I'm interested in starting a discussion about the best way to integrate with GitHub actions: it seems like it would be amazing to have some kind of Tekton + GitHub Actions integration that we officially maintain.

Some ideas that were mentioned at the governing board meeting today:
* We maintain a GitHub action that lets you connect to your Tekton cluster and execute something (would be interesting b/c you wouldn't need triggering?)
* GitHub action that wraps tkn (not exclusive with the above)
* Github action that encapsulates Tekton entirely ( @garethr made a cool proof of concept of this https://garethr.dev/2019/08/running-tekton-in-a-github-action/)

**Comment by afrittoli** [comment]
Thanks for starting this! 

Other ideas:

- Using the GitHub Actions syntax as a DSL on top of Tekton. There was an [initial effort](https://github.com/triggermesh/aktion) by triggermesh on this.
- Write a [GitHub Actions runner](https://github.com/actions/runner) which translates a github action into one or more tekton resources on the fly and executes them against a tekton cluster; it should also include log streaming and all the features the existing runner has.

**Reply by chmouel** [reply]
> Using the GitHub Actions syntax as a DSL on top of Tekton. There was an initial effort by triggermesh on this.

Maybe as an alternative to a DSL on top of tekton is a transpiler converting directly GitHUB actions to tekton?

I think the use case we had for supporting multi arch (on s390x but I can't find the issue anymore cc @barthy1 ) is a good one,  let me try to phrase it in a agiley way : 

As a developer, I would like from my Tekton pipeline build and test a binary for my software on multiple platforms. 
I only have access to Linux hosts from my Tekton pipeline and don't have the infrastructure to spin up OSX VMs. 
I would love to be able to leverage Github actions native way to spin up [Macosx based ](https://docs.github.com/en/desktop/installing-and-configuring-github-desktop/supported-operating-systems#about-supported-operating-systems) VM during my pipeline. 

How can I run a Github action with a specific working, collect the results and make it available to my pipeline ?

**Comment by barthy1** [comment]
> I think the use case we had for supporting multi arch (on s390x but I can't find the issue anymore cc @barthy1 ) is a good one

The discussion was in the PR https://github.com/tektoncd/plumbing/pull/603
Initial suggestion to have multi-arch builds was to use Github actions, but final solution was with dind sidecar and native tekton setup, see this [Task](https://github.com/tektoncd/pipeline/blob/master/tekton/build-push-ma-base-image.yaml ).
